internal/bminventory/inventory.go

```diff
@@ -5437,6 +5456,20 @@ func (b *bareMetalInventory) V2UpdateHostInternal(ctx context.Context, params in
 		}
 	}()
 
+	// Get the bound cluster id out of the transaction.  Note: transaction is not used to query the cluster id
+	if clusterID, err = b.getClusterIDFromHost(params.HostID, params.InfraEnvID); err != nil {
```

Comment:
this function is using `b.db` wouldn't that be a problem because you are in a middle of transaction?

Reply:
Changed